msccl/language/collectives.py

```diff
@@ -62,9 +62,10 @@ def check(self, prog):
 
 
 class AllGather(Collective):
-    def __init__(self, num_ranks, chunk_factor, inplace):
+    def __init__(self, num_ranks, chunk_factor, inplace, create_all_chunks=False):
```

Comment:
Maybe we don't need `create_all_chunks`? For all-gather, in-place case, we can assume the output buffer is already allocated.

Reply:
Maybe add some comments for this. In future we will initialize all input/output buffer

Reply:
Sure! I added some comments, please let me know if they are sufficient or if additional comments are needed.